opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -483,7 +484,19 @@ public void accept(String sinceRevision, Consumer<String> visitor) throws Histor
         }
     }
 
+    @Nullable
+    @Override
+    public HistoryEntry getLastHistoryEntry(File file, boolean ui) throws HistoryException {
+        History hist = getHistory(file, null, null, 1);
+        return getLastHistoryEntry(hist);
+    }
+
     public History getHistory(File file, String sinceRevision, String tillRevision) throws HistoryException {
+        return getHistory(file, sinceRevision, tillRevision, null);
+    }
+
+    public History getHistory(File file, String sinceRevision, String tillRevision,
+                              Integer numCommits) throws HistoryException {
```

Comment:
Maybe add check to the start of the method that if `numCommits` is `null` or `<= 0`, then return `null` or empty? Now it seems that it would return 1 entry even when 0 would be passed.

Reply:
good catch, yes.